include/seqan3/alignment/pairwise/alignment_configurator.hpp

```diff
@@ -440,37 +416,75 @@ struct alignment_configurator
      */
     template <typename function_wrapper_t, typename ...policies_t, typename config_t>
     static constexpr function_wrapper_t configure_free_ends_optimum_search(config_t const & cfg);
+
+    /*!\brief Configures the scoring scheme to use for the alignment computation.
+     *
+     * \tparam function_wrapper_t The invocable alignment function type-erased via std::function.
+     * \tparam config_t The alignment configuration type.
+     *
+     * \param[in] cfg The passed configuration object.
+     *
+     * \returns the configured alignment algorithm.
+     *
+     * \details
+     *
+     * The correct scoring scheme is selected based on the vectorisation mode. If no vectorisation is enabled, the
+     * scoring scheme is the one configured in seqan3::align_config::scoring. If vectorisation is enabled, than the
+     * appropriate scoring scheme for the vectorised alignment algorithm is selected. This involves checking whether the
+     * passed scoring scheme is a matrix or a simple scoring scheme, which has only mistmatch and match costs.
+     */
+    template <typename function_wrapper_t, typename config_t>
+    static constexpr function_wrapper_t configure_scoring_scheme(config_t const & cfg);
+
+    /*!\brief Constructs the actual alignment algorithm wrapped in the passed std::function object.
+     *
+     * \tparam function_wrapper_t The invocable alignment function type-erased via std::function.
+     * \tparam policies_t A template parameter pack for the already configured policy types.
+     * \tparam config_t The alignment configuration type.
+     *
+     * \param[in] cfg The passed configuration object.
+     *
+     * \returns the configured alignment algorithm.
+     *
+     * \details
+     *
+     * Configures the matrix and the gap policy and constructs the algorithm with the configured policies.
+     */
+    template <typename function_wrapper_t, typename ...policies_t, typename config_t>
+    static constexpr function_wrapper_t make_algorithm(config_t const & cfg)
+    {
+        using traits_t = alignment_configuration_traits<config_t>;
+        using matrix_policy_t = typename select_matrix_policy<traits_t>::type;
+        using gap_policy_t = typename select_gap_policy<traits_t>::type;
+
+        return alignment_algorithm<config_t, matrix_policy_t, gap_policy_t, policies_t...>{cfg};
+    }
 };
 
 //!\cond
+template <typename function_wrapper_t, typename config_t>
+constexpr function_wrapper_t alignment_configurator::configure_scoring_scheme(config_t const & cfg)
+{
+    using traits_t = alignment_configuration_traits<config_t>;
+
+    using alignment_scoring_scheme_t =
+        lazy_conditional_t<traits_t::is_vectorised,
+                           lazy<simd_match_mismatch_scoring_scheme,
```

Comment:
This lazy type is something I haven't yet understood (I checked the docs)...
I would be happy if we can talk about it tomorrow in person.